ui/src/app/pages/data/data-set/dataset-page.spec.tsx

```diff
@@ -67,15 +67,24 @@ describe('DataSetPage', () => {
         .find('input').first();
     condition_concept.simulate('change');
     await waitOneTickAndUpdate(wrapper);
-    expect(wrapper.find('[data-test-id="value-list-items"]').length).toBe(2);
+    let valueListItems = wrapper.find('[data-test-id="value-list-items"]');
+    expect(valueListItems.length).toBe(2);
+    let selectedValue = valueListItems.forEach(value =>
+        value.props().checked)
+
+    expect(selectedValue.length).toBe(valueListItems.length);
 
     // Second Concept set in concept set list has domain "Measurement"
     const measurement_concept = wrapper.find('[data-test-id="concept-set-list-item"]').at(1)
         .find('input').first();
     measurement_concept.simulate('change');
     await waitOneTickAndUpdate(wrapper);
     await waitOneTickAndUpdate(wrapper);
-    expect(wrapper.find('[data-test-id="value-list-items"]').length).toBe(5);
+    valueListItems = wrapper.find('[data-test-id="value-list-items"]');
+    selectedValue = valueListItems.forEach(value =>
+        value.props().checked);
+    expect(valueListItems.length).toBe(5);
+    expect(selectedValue.length).toBe(5);
```

Comment:
(no action required) Just for my own understanding: is this the critical assertion that would have failed before the changes introduced in this PR?

And, thought experiment: are there any other edge cases related to the issue addressed by this PR that aren't covered by unit tests here?

It's definitely easier to protect nuanced, complex behavior with a unit test specifically targeted at the given user flow. It might be interesting to consider that for this case, if there's risk that this core test case isn't comprehensive enough.